### PR TITLE
Upgrade to Rundeck 4.17.6

### PR DESCRIPTION
**Pull request by chaopanubo**
Andrew's PR for bug fix had been merged by the code owner.

### CODE REVIEW

**Review by macropin — CHANGES_REQUESTED** [review]
Rundeck does not support JDK17. Only Java 11 is supported. Ref: https://docs.rundeck.com/docs/administration/install/system-requirements.html

File: Dockerfile [hunk 1]
```diff
@@ -102,27 +81,23 @@ RUN set -x \
...
   && cd src \
-  && pip install -r requirements.txt \
+  && python3 -m venv .venv && .venv/bin/pip install --no-cache-dir -U pip setuptools \
+  && .venv/bin/pip install --no-cache-dir -r requirements.txt \
+  && rm requirements.txt \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/* \
   && cp -a /tmp/k8s-sidecar/src/ /sidecar \
   && rm -rf /tmp/k8s-sidecar \
   ;
```
Comment (macropin) [inline Dockerfile]: Please pin this install to a specific release. eg `git checkout 1.25.3`

File: install-openjdk-11.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+set -e 
+
+export DEBIAN_FRONTEND=noninteractive
+
+echo "deb http://deb.debian.org/debian bullseye main" > /etc/apt/sources.list.d/openjdk-11.list
```
Comment (trnubo) [inline install-openjdk-11.sh]: What is this script doing? Are we just installing the debian 11 (bullseye) packing on debian 12 (bookworm)? This is probably not safe.

Comment (trnubo) [reply install-openjdk-11.sh]:
```diff
diff --git a/Dockerfile b/Dockerfile
index d0600dd..793c60f 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -12,10 +12,14 @@ RUN set -x \
   ;

 # Install JDK11
-COPY install-openjdk-11.sh /
 RUN set -x \
   && mkdir /etc/ssl/certs/java/ \
-  && /install-openjdk-11.sh \
+  && wget -O - https://packages.adoptium.net/artifactory/api/gpg/key/public | apt-key add - \
+  && echo "deb https://packages.adoptium.net/artifactory/deb $(awk -F= '/^VERSION_CODENAME/{print$2}' /etc/os-release) main" | tee /etc/apt/sources.list.d/adoptium.list \
+  && apt-get update \
+  && apt-get install --no-install-recommends --no-install-suggests -y temurin-11-jdk \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/* \
   ;

 # Install Google Cloud SDK
```

@macropin what do you think about using adoptium instead? I'll push this to the branch/PR if you agree.

https://adoptium.net/blog/2021/12/eclipse-temurin-linux-installers-available/

File: install-tools.sh [hunk 3]
```diff
@@ -189,66 +189,53 @@ KUBECTL_1_25_8_CHECKSUM_X86_64=80e70448455f3d19c3cb49bd6ff6fc913677f4f240d368fa2
...
+install_crane 0.18.0
 
-install_helm 3.6.3
-install_helm 3.7.2
-install_helm 3.8.2
 install_helm 3.9.4
 install_helm 3.10.3
 install_helm 3.11.2
+install_helm 3.12.3
+
```
Comment (trnubo) [inline install-tools.sh:249]: Add Helm 3.13.x (what ever the latest 3.13 is)

File: install-tools.sh [hunk 4]
```diff
@@ -189,66 +189,53 @@ KUBECTL_1_25_8_CHECKSUM_X86_64=80e70448455f3d19c3cb49bd6ff6fc913677f4f240d368fa2
...
 
-install_crane 0.15.2
+install_crane 0.18.0
 
-install_helm 3.6.3
-install_helm 3.7.2
-install_helm 3.8.2
 install_helm 3.9.4
 install_helm 3.10.3
 install_helm 3.11.2
```
Comment (trnubo) [inline install-tools.sh]: Upgrade to 3.11.3

File: install-tools.sh [hunk 5]
```diff
@@ -189,66 +189,53 @@ KUBECTL_1_25_8_CHECKSUM_X86_64=80e70448455f3d19c3cb49bd6ff6fc913677f4f240d368fa2
...
+install_helm 3.12.3
+
 
-install_kubectl 1.21.3
-install_kubectl 1.22.4
-install_kubectl 1.23.17
 install_kubectl 1.24.12
 install_kubectl 1.25.8
 install_kubectl 1.26.3
```
Comment (trnubo) [inline install-tools.sh]: Upgrade to the latest 1.24.x, 1.25.x and 1.26.x releases

Add 1.27 and 1.28 latest releases

**Comment by trnubo** [comment]
Please update this PR.

We want to:
- Stick to debian:11 upstream
- Install AWS CLI v1 via pip instead of apt-get
- Stick to rundeck 4.x versions
- Check all tools are upto date
- Include k8s 1.29
- Include helm 3.14.x

File: Dockerfile [hunk 6]
```diff
@@ -108,22 +108,23 @@ RUN set -x \
...
-  && pip install -r requirements.txt \
+  && pip install --no-cache-dir -r requirements.txt \
+  && rm requirements.txt \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/* \
   && cp -a /tmp/k8s-sidecar/src/ /sidecar \
   && rm -rf /tmp/k8s-sidecar \
   ;
-
+
```
Comment (macropin) [inline Dockerfile]: unnecessary whitespace

**Review by macropin — CHANGES_REQUESTED** [review]
Other than that whitespace change. LGTM

File: Dockerfile (hunk 6 above)
Comment (chaopanubo) [reply Dockerfile]: Fixed.

**Comment by chaopanubo** [comment]
> Other than that whitespace change. LGTM

Thank you, Andrew.

**Pull request merged**